scripts/spack_manifest/injection/modules.py

```diff
@@ -0,0 +1,279 @@
+            yaml.dump(
+                manifest_with_projections_and_includes,
+                output_file,
+                default_flow_style=False,
+                sort_keys=False,
+            )
+
+def inject_projections(
+    manifest: str, root_spec: str, packages: set[str]
+) -> dict[str, Any]:
+
+    # Get projections that are already defined in the manifest - we don't want to redefine these
+    defined_projections: dict[str, str] = _get_defined_projections(manifest)
+    defined_projections_set: set[str] = set(defined_projections.keys())
+    # Get packages that have versions defined in the manifest - we can only generate projections for things with an explicit version
+    packages_with_versions_defined: set[str] = _get_packages_with_versions_defined(
+        manifest
+    )
+
+    # Essentially...we want to generate projections for all packages that don't already have them, provided they have a version defined
+    projections_to_generate: set[str] = (packages & packages_with_versions_defined) - defined_projections_set
+
+    # To start with, add the projections that are already defined in the manifest
+    new_projections: dict[str, str] = dict(defined_projections)
+
+    if root_spec not in defined_projections_set:
+        new_projections.update(generate_projection_for_root_spec_or_raise(manifest, root_spec))
+
+    for projection in projections_to_generate:
+        new_projections.update(
+            generate_projection_for_package_or_raise(manifest, projection)
+        )
+
+    # Sort the projections by name to ensure a consistent order...
+    ordered_new_projections = dict(sorted(new_projections.items()))
+
+    # But, we want to ensure that the root spec projection is always first in the list because it's special
+    root_spec_version = ordered_new_projections.pop(root_spec)
+    ordered_new_projections = {root_spec: root_spec_version, **ordered_new_projections}
+
+    # Finally, add the new projections to the manifest
+    injected_manifest: dict[str, Any] = dict(manifest)
+    injected_manifest.setdefault("spack", {}).setdefault("modules", {}).setdefault("default", {}).setdefault("tcl", {})["projections"] = ordered_new_projections
+
+    return injected_manifest
+
+def inject_includes(
+    manifest: dict[str, Any], root_spec: str, packages: set[str]
+) -> dict[str, Any]:
+    # We want to inject the includes for the root spec, all packages defined in --packages, and existing includes in the manifest.
+    existing_includes: set[str] = _get_defined_includes(manifest)
+
+    # Includes are the union of packages, and existing includes - we add the root spec later
+    includes: set[str] = packages | existing_includes
+
+    # To sort, we want to ensure that the root spec is always first in the list, and the rest are sorted alphabetically
+    sorted_include: list[str] = [root_spec] + sorted(includes)
+
+    # Finally, we inject the includes into and updated manifest, which we return
```

Comment:
```suggestion
    # Finally, inject the includes into a new copy of the manifest, which is returned
```
Just stressing that it is a new, deep, copy of the manifest